framework/src/executioners/EigenExecutionerBase.C

```diff
@@ -205,10 +199,15 @@ EigenExecutionerBase::inversePowerIteration(unsigned int min_iter,
   mooseAssert(tol_eig>0.0, "Invalid eigenvalue tolerance");
   mooseAssert(tol_x>0.0, "Invalid solution norm tolerance");
 
-  if ((_bx_execflag & (EXEC_TIMESTEP_END | EXEC_LINEAR)) == EXEC_NONE)
-    mooseError("rhs postprocessor for the power method has to be executed on timestep or residual");
-  if ((_xdiff_execflag & (EXEC_TIMESTEP_END | EXEC_LINEAR)) == EXEC_NONE)
-    mooseError("xdiff postprocessor for the power method has to be executed on timestep or residual");
+  // obtain the solution diff
+  const PostprocessorValue * solution_diff = NULL;
+  if (xdiff != "")
+  {
+    solution_diff = &getPostprocessorValueByName(xdiff);
+    ExecFlagType xdiff_execflag = _problem.getUserObject<UserObject>(xdiff).execBitFlags();
+    if ((xdiff_execflag & EXEC_LINEAR) == EXEC_NONE)
+      mooseError("solution diff postprocessor must be evaluated on linear currently");
```

Comment:
The power method algorithm basically says the solution differences (also the eigenvalue) can be evaluated after each power iteration. We currently use postprocessors for evaluating solution differences and eigenvalue. The old implementation lets the postprocessors be evaluated on timestep_end and call auxkernel and uo compute after each power iteration. This causes issues when some other auxkernels/uos cannot be evaluated after each power iteration. So I decided to make those postprocessors for power iteration always on linear. Although in this way we will introduce more unnecessary evaluations, but we do not need to evaluate auxkernels/uos after power iteration thus avoid the issues. If we have a new execute_on flag, like EXEC_STEP, then we can let those postprocessors on it.